diff --git a/common_drivers/drivers/media/di_multi/di_dbg.h b/common_drivers/drivers/media/di_multi/di_dbg.h
index e6cd6b35b..417a60b2c 100644
--- a/common_drivers/drivers/media/di_multi/di_dbg.h
+++ b/common_drivers/drivers/media/di_multi/di_dbg.h
@@ -22,6 +22,7 @@
 #include <linux/amlogic/media/vfm/vframe.h>
 #include <linux/amlogic/media/vfm/vframe_provider.h>
 #include <linux/amlogic/media/vfm/vframe_receiver.h>
+#include "di_data_l.h"
 
 #define CRC_COUNT_NUB		(20)
 #define CRC_NUB		(3)
@@ -42,7 +43,7 @@ void ddbg_reg_save(unsigned int addr, unsigned int val,
 void dim_ddbg_mod_save(unsigned int mod,
 		       unsigned int ch,
 		       unsigned int cnt);
-void ddbg_sw(unsigned int mode, bool on);
+void ddbg_sw(enum EDI_LOG_TYPE mode, bool on);
 
 /********************************
  *time:
diff --git a/common_drivers/drivers/media/dtv_demod/dvb_qam/dvbc_func.h b/common_drivers/drivers/media/dtv_demod/dvb_qam/dvbc_func.h
index b5aeb7a3d..0d4cf108f 100644
--- a/common_drivers/drivers/media/dtv_demod/dvb_qam/dvbc_func.h
+++ b/common_drivers/drivers/media/dtv_demod/dvb_qam/dvbc_func.h
@@ -467,7 +467,7 @@ enum fe_modulation amdemod_qam_fe(enum qam_md_e qam);
 int dvbc_get_power_strength(int agc_gain, int tuner_strength);
 int dvbc_status(struct aml_dtvdemod *demod, struct aml_demod_sts *demod_sts,
 		struct seq_file *seq);
-void demod_dvbc_set_qam(struct aml_dtvdemod *demod, unsigned int qam, bool auto_sr);
+void demod_dvbc_set_qam(struct aml_dtvdemod *demod, enum qam_md_e qam, bool auto_sr);
 void demod_dvbc_store_qam_cfg(struct aml_dtvdemod *demod);
 void demod_dvbc_fsm_reset(struct aml_dtvdemod *demod);
 u32 dvbc_get_status(struct aml_dtvdemod *demod);
diff --git a/common_drivers/drivers/media/enhancement/amvecm/set_hdr2_v0.h b/common_drivers/drivers/media/enhancement/amvecm/set_hdr2_v0.h
index 9c9608375..bc1769b53 100644
--- a/common_drivers/drivers/media/enhancement/amvecm/set_hdr2_v0.h
+++ b/common_drivers/drivers/media/enhancement/amvecm/set_hdr2_v0.h
@@ -81,7 +81,7 @@ enum hdr_hist_sel {
 #define OO_NOR			(OO_BITS - OO_GAIN_SHIFT)
 
 enum hdr_process_sel hdr_func(enum hdr_module_sel module_sel,
-			      u32 hdr_process_select,
+			      enum hdr_process_sel hdr_process_select,
 			      struct vinfo_s *vinfo,
 			      struct matrix_s *gmt_mtx,
 			      enum vpp_index_e vpp_index);
@@ -124,7 +124,7 @@ int hdr10p_ebzcurve_update(enum hdr_module_sel module_sel,
 			   struct hdr10pgen_param_s *hdr10pgen_param,
 			   enum vpp_index_e vpp_index);
 enum hdr_process_sel hdr10p_func(enum hdr_module_sel module_sel,
-				 u32 hdr_process_select,
+				 enum hdr_process_sel hdr_process_select,
 				 struct vinfo_s *vinfo,
 				 struct matrix_s *gmt_mtx,
 				 enum vpp_index_e vpp_index);
diff --git a/common_drivers/drivers/media/enhancement/amvecm/vlock.h b/common_drivers/drivers/media/enhancement/amvecm/vlock.h
index 8b13ad71e..728ce4be4 100644
--- a/common_drivers/drivers/media/enhancement/amvecm/vlock.h
+++ b/common_drivers/drivers/media/enhancement/amvecm/vlock.h
@@ -320,7 +320,7 @@ enum vlock_pll_sel {
 #define ENCL_FRC_CTRL_T3X                    0x14D9
 
 /* 0:enc;1:pll;2:manual pll */
-extern unsigned int vlock_mode;
+extern enum VLOCK_MD vlock_mode;
 extern unsigned int vlock_en;
 extern unsigned int vecm_latch_flag;
 /*extern void __iomem *amvecm_hiu_reg_base;*/
diff --git a/common_drivers/drivers/media/vin/tvin/hdmirx/hdmi_rx_drv_ext.h b/common_drivers/drivers/media/vin/tvin/hdmirx/hdmi_rx_drv_ext.h
index 5aca3bbe2..26ef92c39 100644
--- a/common_drivers/drivers/media/vin/tvin/hdmirx/hdmi_rx_drv_ext.h
+++ b/common_drivers/drivers/media/vin/tvin/hdmirx/hdmi_rx_drv_ext.h
@@ -6,6 +6,8 @@
 #ifndef __HDMI_RX_DRV_EXT_H__
 #define __HDMI_RX_DRV_EXT_H__
 
+#include <linux/amlogic/media/frame_provider/tvin/tvin.h>
+
 /* for CEC notify */
 typedef int (*cec_callback)(unsigned int pwr5v_sts);
 typedef void (*cec_spd_callback)(int port_id, int handle_type);
@@ -18,5 +20,5 @@ int register_cec_callback(cec_callback callback);
 void unregister_cec_callback(void);
 unsigned int hdmirx_get_base_fps(unsigned int hw_vic);
 unsigned int hdmirx_get_hpd_info(void);
-void rx_update_edid_callback(u32 tvin_port, u32 hdr_priority);
+void rx_update_edid_callback(enum tvin_port_e tvin_port, u32 hdr_priority);
 #endif
